pandas/core/reshape/reshape.py

```diff
@@ -715,71 +716,241 @@ def _convert_level_number(level_num, columns):
     return result
 
 
-@Appender(_shared_docs['melt'] %
-          dict(caller='pd.melt(df, ',
-               versionadded="",
-               other='DataFrame.melt'))
-def melt(frame, id_vars=None, value_vars=None, var_name=None,
-         value_name='value', col_level=None):
+def _melt(frame, id_vars=None, value_vars=None, var_name=None,
+          value_name='value', col_level=None, stubnames=False,
```

Comment:
i would be fine with another module called melt.py
which contained all of the code for melt; 

that way helper functions can be broken out and impl is understandable

Reply:
I created melt.py and test_melt.py. I moved melt out of reshape.py and corrected some imports. I heavily commented the code as well. I do need at least 20 more tests as there is a huge range of possibilities now with this new implementation.